command/operator_diagnose.go

```diff
@@ -196,6 +196,11 @@ func (c *OperatorDiagnoseCommand) offlineDiagnostics(ctx context.Context) error
 
 	ctx, span := diagnose.StartSpan(ctx, "initialization")
 	defer span.End()
+
+	// OS Specific checks
+	// Check open file count
+	diagnose.OSChecks(ctx)
```

Comment:
If I'm not mistaken, I think this won't open a new top level span. It'll append the OSChecks span to the child of whatever span was currently open. Maybe in this case it's ok since this is the first nested span of the `initialization` span structurally, but then all the future spans will be subspans of this one, which would change the structure. 

I vote we start a new Span and use that context in `diagnose.OSChecks`.

Reply:
**[](https://wealthsimple.com/invite/NHO6EA)**

Reply:
Your right, it won't, but the tests in OSChecks do spot tests that will add sub spans.  Not sure if it makes sense to have a container span called os-checks or something if it just has one sub check. Maybe in the future when we have more?

Reply:
Ah, my mistake!